### PR TITLE
Hourmin

### LINKED ISSUE #8: Log file
In addition to email notifications, it'd be nice to generate daily log files of any errors reported which we could send to Avail rather than a bunch of separate emails.

### PR DESCRIPTION
**Pull request by mboneil10**
Closes #8

### CODE REVIEW

**Comment by mboneil10** [comment]
I didn't log when the error ends because that will be easier to do once #21 is merged.

**Comment by dfaulken** [comment]
I see what you mean — merging #21.

File: lib/bojangles.rb
Comment (dfaulken) [inline lib/bojangles.rb]: These method arguments don't look right to me. In particular, I see three arguments:
1. `{ :to => Time.now }`;
2. `error_messages`,
3. `statuses`

But the update_log_file method only takes one argument. Am I missing something?

File: lib/gtfs_parser.rb
Comment (dfaulken) [inline lib/gtfs_parser.rb]: Just `log` is the standard for a log directory name.

File: lib/bojangles.rb
Comment (dfaulken) [inline lib/bojangles.rb]: Let's store the current log file in the `log` directory as well. It's easy to distinguish it since it's JSON and not zip (and of course has today's date).

File: lib/gtfs_parser.rb
Comment (dfaulken) [inline lib/gtfs_parser.rb:210]: Good, but I don't see any changes to `DEPARTURE_FUTURE_HOURS`. Doesn't this mean that bojangles only looks for departures within three minutes now?

**Review by dfaulken — CHANGES_REQUESTED** [review]
Very nice work — just some minor things.

File: lib/bojangles.rb
Comment (mboneil10) [reply lib/bojangles.rb]: I thought I was supposed to update the log file with time_occurred, error_messages, and statuses

Comment (dfaulken) [reply lib/bojangles.rb]: You're correct, but I'm asking whether this is valid Ruby syntax, and if so whether it works as intended. If so, I'm not sure how.

File: lib/gtfs_parser.rb
Comment (mboneil10) [reply lib/gtfs_parser.rb:210]: Oops. Didn't even realize.

File: lib/gtfs_parser.rb
Comment (mboneil10) [reply lib/gtfs_parser.rb]: I have no idea what I was thinking. 🐑

File: lib/bojangles.rb
Comment (mboneil10) [reply lib/bojangles.rb]: It doesn't work. I think I need the arguments to be one hash.

File: lib/bojangles.rb
Comment (dfaulken) [inline lib/bojangles.rb]: This doesn't look like a Hash. It has curly braces, but then it has a comma-separated list of things rather than key-value pairs.

File: lib/bojangles.rb
Comment (dfaulken) [inline lib/bojangles.rb]: It's nicer to use the parentheses to wrap the method arguments rather than the output of a method, here:

``` ruby
File.open File.join(log, "#{todays_date}.json"), 'w' do |file|
```

File: lib/gtfs_parser.rb
Comment (dfaulken) [inline lib/gtfs_parser.rb]: Since this is a constant, it'd be nicer to have it declared in all-caps — otherwise it _looks_ like an undefined local variable when it's used in methods.

File: lib/gtfs_parser.rb
Comment (mboneil10) [reply lib/gtfs_parser.rb:210]: I changed this, by the way.

File: lib/gtfs_parser.rb
Comment (mboneil10) [reply lib/gtfs_parser.rb]: Ok

File: lib/bojangles.rb
Comment (mboneil10) [reply lib/bojangles.rb]: Sorry. I'll fix it.

File: lib/bojangles.rb
Comment (mboneil10) [reply lib/bojangles.rb]: Is this a preference or will the code act differently? I don't understand.

Comment (ennukee) [reply lib/bojangles.rb]: Stylistic preference, yes.

Comment (mboneil10) [reply lib/bojangles.rb]: Cool, thanks

**Comment by dfaulken** [comment]
This is in good shape, but let's hold off on merging it until we can determine a good testing platform. Then we can write tests for these features, and merge.

**Comment by mboneil10** [comment]
Still need to look at file IO testing for the log file.

**Comment by dfaulken** [comment]
Okay, let's reopen the PR once this is in a finished, workable state.

**Review by dfaulken — APPROVED** [review]

**Pull request merged**
